### ISSUE TITLE
Query on Raspberry Pi

### ISSUE THREAD

**Issue by CreativeWolf** [seemoo-lab/mobisys2018_nexmon_software_defined_radio#2]
Hi,

Can this patch be applied for BCM43143 that's on Raspberry Pi 3?

Thanks

**Comment by matthiasseemoo** [comment]
No, but to the wifi chip of the new rpi3+ after some adjustments. The only
thing you need is an 802.11ac phy from broadcom.

Am 15.03.2018 9:13 nachm. schrieb "CreativeWolf" <notifications@github.com>:

**Comment by F5OEO** [comment]
As a brief understanding, I/Q are sent through ioctl to RAM. Is it possible to feed I/Q samples continuously. If yes, what could be the samplerate which can be achieved ? 
Thanks for this project which could open large SDR applications.

**Comment by matthiasseemoo** [comment]
Not just to RAM but to template ram, which is part of the d11 core. Writing
to template ram directly from the arm core is quite slow, as the template
ram is not memorymapped into the arm's address space and you have to write
through a register interface.

There is however the dma controller used to send wifi frames to the d11
core. I did not figure out in detail how to configure it, but i observed
that the first few transmitted frames end up in template ram, but this
specific section is never overwritten by new frames, only when we restart
the d11 core. Hence, continuous transmissions require more reverse
engineering and research.

Additionally the minimum sdr tx sample rate is 40 MSps, with 32 bit per
sample, which means 160 MBps or 1280 Mbps transferrate into the wifi chip,
which is quite high and likely not supported by sdio connected chips. So we
need pcie connected chips such as the bcm4358 on a nexus 6p.

Am 16.03.2018 11:38 schrieb "F5OEO" <notifications@github.com>:

**Comment by E3V3A** [comment]
Hi,
Sorry to repeat the question,

>Q: Can this patch be applied for BCM43143 that's on Raspberry Pi 3?
> A: No, but to the wifi chip of the new rpi3+ after some adjustments. The only
thing you need is an 802.11ac phy from broadcom.

What is the reason for this? Is it a HW limitation or just that you didn't reverse for the RPi3?

**Comment by matthiasseemoo** [comment]
You need an 802.11ac phy, so its a hw limitation.

Am 16.03.2018 14:11 schrieb "E:V:A" <notifications@github.com>:

**Comment by F5OEO** [comment]
@matthiasseemoo Thank you for all these info. I wait for Rpi3B+ and could maybe use DMA to fill template RAM. However I have surely to wait for a patch of your work on Raspberry. Congrat again for this great work.

**Comment by E3V3A** [comment]
[Here](https://www.cisco.com/c/dam/en/us/products/collateral/wireless/aironet-3600-series/white-paper-c11-713103.pdf) is a very nice document explaining all bout the `802.11ac phy` standard. 

![802 11ac1](https://user-images.githubusercontent.com/194392/37556665-cd1cdb34-2a01-11e8-8cf8-fd1be45b3542.png)

**Comment by garci66** [comment]
Quick question. For the Rpi3b+,  Will it be possible at all to receive the raw I/Q data? or any way of doing proper spectrum analysis on the wifi bands? Thanks!

**Comment by matthiasseemoo** [comment]
Reception should be possible. For spectrum analysis you either need a more
advanced chip or you have to calculate the fft on your own.

Diego Garcia del Rio <notifications@github.com> schrieb am So., 15. Apr.
2018, 06:14:

**Comment by guysoft** [comment]
Hey, quiet follower - FFT should not be a challenge right?

**Comment by garci66** [comment]
Indeed, I think FFT should be manageable. I need to look a bit more in detail on how one would do the receiving, etc. But the initial cofirmation is encouraging.
